subsys/logging/log_core.c

```diff
@@ -478,6 +492,69 @@ u32_t log_filter_get(struct log_backend const *const backend,
 	}
 }
 
+static void *log_malloc(u32_t len)
+{
+	int err;
+	u32_t *buf;
+
+	if (len > CONFIG_LOG_STRDUP_MAX_STRING) {
```

Comment:
The malloc()/free() API does not match well with k_mem_slab_alloc().
As result you have `if (len > CONFIG_LOG_STRDUP_MAX_STRING)` here and `log_malloc(CONFIG_LOG_STRDUP_MAX_STRING)` in log_strdup(). What about removing this abstraction and using slab allocator directly?